research/picai/nnunet_scripts/predict_and_eval.py

```diff
@@ -0,0 +1,252 @@
+            'network_weights' key or nn.Modules. If the path points to a
+            directory it should be an nnunet results folder for a particular
+            dataset-config-trainer combo. The plans key should be the path to
+            the nnunet model plans json file. The dataset_json key should be
+            the path to the dataset json of one of the training datasets. Or
+            create a new json yourself with the 'label' and 'file_ending' keys
+            and their corresponding values as specified by nnunet.
+        inputs_folder (str): Path to the folder containing the raw input data
+            that has not been processed by nnunet yet. File names must follow
+            the nnunet convention where each channel modality is stored as a
+            seperate file. File names should be case-identifier_0000 where
+            0000 is a 4 digit integer representing the channel/modality of the
+            image. All cases must have the same N channels numbered from 0 to
+            N.
+        labels_folder (str): Path to the folder containing the ground truth
+            annotation maps. File names must match the case identifiers of the input images
+        output_folder (Optional[str], optional): Path to the output folder. By
+            default the only output is a 'metrics.json' file containing the
+            evaluation results. If left as none then nothing is saved.
+            Defaults to None.
+        save_probability_maps (bool, optional): Whether or not to save the
+            predicted probability maps. Defaults to False.
+        save_detection_maps (bool, optional): Whether or not to save the
+            predicted lesion detection maps. Defaults to False.
+        verbose (bool, optional): Whether or not to print logs to stdout.
+            Defaults to True.
+    """
+
+    # Ensure an output folder is specified
+    if save_probability_maps or save_detection_maps or save_annotations:
+        assert (
+            output_folder is not None
+        ), "Can not save the probability maps, annotations \
+            or the detection maps if no output folder is specified"
+
+    # Run inference and maybe save probability maps.
+    # output folders will be created if they dont exist
+    if save_probability_maps and output_folder is not None:
+        prob_out_folder = join(output_folder, "predicted_probability_maps")
+    else:
+        prob_out_folder = None
+
+    if save_annotations and output_folder is not None:
+        annotations_folder = join(output_folder, "predicted_annotations")
+    else:
+        annotations_folder = None
+
+    probability_preds, annotation_preds, case_identifiers = predict(
+        config_path=config_path,
+        input_folder=inputs_folder,
+        probs_folder=prob_out_folder,
+        annotations_folder=annotations_folder,
+        verbose=verbose,
+    )
+    del annotation_preds
+
+    # Extract lesion detection maps
+    if verbose:
+        log(INFO, "")
```

Comment:
Is this just to create seperation or something?

Reply:
Yep